### PR TITLE
Fixes for local development

### PR DESCRIPTION
**Pull request by mdp**
A few issues that I ran into getting this fully up on my local machine.
I also made sure it works now properly on Github Codespaces which
will hopefully make it easier for devs to get started.

- Username/Pass docs updated to correct values (was factory@factory, 
now firezone@localhost)
- Add runtime.exs and migrate releases.exs over to it as recommended by
Elixir 1.11 release notes
- Update Endpoint to allow for proxy passed headers if enabled with
PROXY_FORWARDED
- Add missing .env file section to CONTRIBUTION
- Add a sample .env file (.env.sample) to help getting started
- Update docs to show how to use Github Codespaces with an external
URL.
- Remove unused alias from dev (which I think doesn't work anyway due to
compile time restrictions, only in runtime or releases.exs)
- Ensure pre-commit is in path and available on devcontainer
- Simplify some docker commands, remove unneeded ports.

### CODE REVIEW

**Comment by mdp** [comment]
The biggest change in this is the move from releases.exs to runtime.exs. I've not attempted a release process yet, just the basic tests.

**Comment by jamilbk** [comment]
@mdp Thanks for cleaning up all these cobwebs. I've added you to the repo with Write access so the CI pipeline should kickoff on your pushes now (that'll at least smoke test runtime.exs).

File: apps/fz_http/lib/fz_http_web/endpoint.ex [hunk 1]
```diff
@@ -2,6 +2,10 @@ defmodule FzHttpWeb.Endpoint do
   use Phoenix.Endpoint, otp_app: :fz_http
   alias FzHttpWeb.Session
 
+  if Application.get_env(:fz_http, FzHttpWeb.Endpoint, :proxy_forwarded) do
```
Comment (jamilbk) [inline apps/fz_http/lib/fz_http_web/endpoint.ex:5]: I think this may also fix #480 🎉

**Review by jamilbk — APPROVED** [review]
This is great -- this brings us a step closer to developing with a fully functional local Firezone on any platform (the fz_wall and fz_vpn applications are configured in `:dev` envs to stub the `wg` and `nft` commands and return dummy data).

There are a couple things I want to clean up, but I'll go ahead and merge this and open a new PR for those.

**Comment by mdp** [comment]
Yeah, let me know if there's anything I can improve. I'm very new to Elixir. My first goal was to get up a copy of Firezone in a reproducible environment, and that part is mostly done.

With regards to nft and wg, we can install these inside the docker container, but it probably makes sense to have them stubbed in dev for now for simplicity.

**Comment by mdp** [comment]
Closing in favor of #562

**Pull request closed without merging**